### PR TITLE
feat(wallet-connect-2): lazy loading for wc1 and wc2

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

The key difference here is that `@walletconnect` bundle loads only on demand, before this fix it was loaded always.
The result: -238kb from page loading.

Before:
image

After:
image

  # To Test

1. Wallet connect v1
 - connect wallet (all networks)
 - place order
 - reload page with connected wallet (should restore connection)
2. Wallet connect v2
 - connect wallet (all networks)
 - place order
  - reload page with connected wallet (should restore connection)

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/wallet/web3-react/connection/asyncConnector.ts [hunk 1]
```diff
@@ -1,27 +1,50 @@
+import { Web3Provider } from '@ethersproject/providers'
 import { Actions, Connector } from '@web3-react/types'
 
+import EventEmitter from 'events'
+
+export const ASYNC_CUSTOM_PROVIDER_EVENT = 'customProvider'
+
+function initCustomProvider(self: AsyncConnector, connector: Connector, chainId: number) {
+  if (connector.provider && !connector.customProvider) {
+    self.customProvider = new Web3Provider(connector.provider, chainId)
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/wallet/web3-react/connection/asyncConnector.ts:10]: Just inherited it from:
https://github.com/Uniswap/web3-react/blob/main/packages/core/src/hooks.ts#L366

File: apps/cowswap-frontend/src/modules/wallet/web3-react/connection/asyncConnector.ts [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
+  }
+
+  self.events.emit(ASYNC_CUSTOM_PROVIDER_EVENT, self.customProvider)
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/wallet/web3-react/connection/asyncConnector.ts:13]: Explicitly fire an event when provider from AsyncConnector was loaded

**Review by alfetopito — APPROVED** [review]
Was able to trade using both wc v1 and v2

**Comment by anxolin** [comment]
Awesome!

**Pull request merged**